test/new-e2e/tests/installer/package_agent_test.go

```diff
@@ -38,19 +53,247 @@ func (s *packageAgentSuite) TestInstall() {
 	// state.AssertFileExists("/etc/datadog-agent/install.json", 0644, "dd-agent", "dd-agent")
 }
 
-func (s *packageAgentSuite) TestExperimentStartedButNotInstalled() {
+func (s *packageAgentSuite) TestTimeout() {
+	s.RunInstallScript(envForceInstall("datadog-agent"))
+	defer s.Purge()
+	s.host.WaitForUnitActive("datadog-agent.service", "datadog-agent-trace.service", "datadog-agent-process.service")
+
+	s.host.SetupFakeAgentXP()
+
+	// assert timeout is already set
+	s.host.AssertUnitProperty("datadog-agent-exp.service", "JobTimeoutUSec", "50min")
+
+	// shorten timeout for tests
+	s.host.Run("sudo mkdir -p /etc/systemd/system/datadog-agent-exp.service.d/")
+	defer s.host.Run("sudo rm -rf /etc/systemd/system/datadog-agent-exp.service.d/")
+	s.host.Run(`echo -e "[Unit]\nJobTimeoutSec=5" | sudo tee /etc/systemd/system/datadog-agent-exp.service.d/override.conf > /dev/null`)
+	s.host.Run(`sudo systemctl daemon-reload`)
+
+	s.host.AssertUnitProperty("datadog-agent-exp.service", "JobTimeoutUSec", "5s")
+
+	timestamp := s.host.LastJournaldTimestamp()
+	s.host.Run(`sudo systemctl start datadog-agent-exp --no-block`)
+
+	s.host.AssertSystemdEvents(timestamp, host.SystemdEvents().
+		// first stop agent dependency
+		Unordered(host.SystemdEvents().
+			Stopped(traceUnit).
+			Stopped(processUnit),
+		).
+		// then agent
+		Stopping(agentUnit).
+		Stopped(agentUnit).
+
+		// start experiment dependency
+		Unordered(host.SystemdEvents().
+			Starting(agentUnitXP).
+			Started(processUnitXP).
+			Started(traceUnitXP).
+			Skipped(probeUnitXP).
+			Skipped(securityUnitXP),
+		).
+
+		// timeout
+		Timed(agentUnitXP).
+		Unordered(host.SystemdEvents().
+			Stopped(agentUnitXP).
+			Stopped(processUnitXP).
+			Stopped(traceUnitXP),
+		).
+
+		// start stable agents
+		Started(agentUnit).
+		Unordered(host.SystemdEvents().
+			Started(traceUnit).
+			Started(processUnit).
+			Skipped(probeUnit).
+			Skipped(securityUnit),
+		),
+	)
+}
+
+func (s *packageAgentSuite) TestSigkill() {
 	s.RunInstallScript(envForceInstall("datadog-agent"))
 	defer s.Purge()
 	s.host.WaitForUnitActive("datadog-agent.service", "datadog-agent-trace.service", "datadog-agent-process.service")
 
+	s.host.SetupFakeAgentXP().
+		SetStopWithSigkill("core-agent").
+		SetStopWithSigkill("process-agent").
+		SetStopWithSigkill("trace-agent")
+
+	for _, unit := range []string{traceUnitXP, processUnitXP, agentUnitXP} {
```

Comment:
Nit: You're missing some `s.Log` at the start of a bunch of those `for ... []string{testCases}` pattern. It's a pain to debug which one you are in (which I assume you have experienced since you added it bellow)

Reply:
good catch, this is the only one missing